documentation/docs/steps/cloudFoundryCreateServiceKey.md

```diff
@@ -23,12 +23,12 @@ The following example creates a service key named "myServiceKey" for the service
 cloudFoundryCreateServiceKey(
   script: this,
   cloudFoundry: [
-      apiEndpoint: 'https://test.server.com',
-      credentialsId: 'cfCredentialsId',
-      org: 'cfOrg',
-      space: 'cfSpace',
-      serviceInstance: 'myServiceInstance',
-      serviceKeyName: 'myServiceKey',
-      serviceKeyConfig: '{ \"key\" : \"value\" }'
+      cfApiEndpoint: 'https://test.server.com',
+      cfCredentialsId: 'cfCredentialsId',
+      cfOrg: 'cfOrg',
+      cfSpace: 'cfSpace',
+      cfServiceInstance: 'myServiceInstance',
+      cfServiceKeyName: 'myServiceKey',
+      cfServiceKeyConfig: 'serviceKeyConfig.json'
```

Comment:
One suggestions: you could also change the name of the parameter. Maybe 'cfServiceKeyConfigFile' to make it clear that a file name is expected now.

Reply:
I think changing it to `cfServiceKeyConfigFile` and switching it back to `cfServiceKeyConfig` would also be confusing since it is planned to implement in-line passing of a JSON string in the near future.

Reply:
I would not overload `cfServiceKeyConfig` with the option to pass either the in-line string or the filename. You could use separate parameters for both options, but only one of those parameters needs to be passed to the function. Hence my suggestion.

Reply:
To me overloading (... passing a json text OR passing a file) looks also confusing. Hence I personally think having two parameters for that is the better way. How does the underlying `cf` command line tool behave? Does it accept either a file name OR a json string. In case that is true that would be a point for doing it here in the same way.

Reply:
The cf cli commands look like this:

- cf create-service-key SERVICE_INSTANCE SERVICE_KEY -c '{"name":"value","name":"value"}'

- cf create-service-key SERVICE_INSTANCE SERVICE_KEY -c PATH_TO_FILE

Reply:
> The cf cli commands look like this: [...]

so `-c` can carry either a file or a json-string? In that case we can argue we simply mimic the behaviour of the tool. Looks confusing to me nevertheless.

Reply:
Yes, `-c` can carry both the path to a file as well as a JSON string.

However when passing a file you need to run the parameter as follows:
`cf create-service-key testInstance testKey -c testFile.json`

When passing a JSON string it has to be a JSON wrapped in upper quotes:
`cf create-service-key testInstance testKey -c '{"name":"value","name":"value"}'`

In that case, when passing the argument in upper quotes with `cmd.RunExecutable()` the `cf` command fails. The command gets passed to `cmd.RunExecutable()` function analog to the official Cloud Foundry documentation in the current release without this fix but still doesn't work.
Passing a config file works fine, that's why this fix only covers passing config file as the only supported option right now.

More investigation needs to be done on why this issue occurs when passing a valid JSON config so both options will be available. I think reducing the native cf cli function to only passing a JSON config file forever doesn't make sense.

Reply:
> However when passing a file you need to run the parameter as follows:
> cf create-service-key testInstance testKey -c testFile.json

> When passing a JSON string it has to be a JSON wrapped in upper quotes:
> cf create-service-key testInstance testKey -c '{"name":"value","name":"value"}'

But I guess on the command line 

`cf create-service-key testInstance testKey -c 'testFile.json'`

works also (mind the quotes surrounding `testFile.json`. Hence for the command line we can ' unify' those two cases by having the quotes ... By the way: I would like to recommand to use the quotes also in the case of having a file, think about having a file with a blank in the name/path.

Everything written above relates to the command line, not to the go coding!

Regarding the go-coding: We have

```
cfCreateServiceKeyScript = []string{"create-service-key", config.CfServiceInstance, config.CfServiceKeyName, "-c", config.CfServiceKeyConfig}
```

I would expect that providing a json via `config.CfServiceKeyConfig` does the job. I will try to run it and check more details ...

Reply:
@dominiklendle : can you provide a working command line (with -c pointing to a file).

I would like to 

1.) run that command with the file
2.) extract the json string and run it with providing the json string.

Reply:
After checking, providing a json via `config.CfServiceKeyConfig` works. Hence, both options are working again after testing.

It seems like the official cf documentation mainly covers cf cli commands via local cli where you need to wrap upper single quotes around the string. 

When passing the arguments as a string array to `cmd.RunExecutable()` function you don't need upper quotes for the JSON config. This is only needed for local cf cli execution for parsing the string. This works in Go:
```
cfCreateServiceKeyScript = []string{"create-service-key", config.CfServiceInstance, config.CfServiceKeyName, "-c", config.CfServiceKeyConfig}
```
I will leave the parameter name at `cfServiceKeyConfig` since both passing a JSON string as well as a config file are working now again. Documentation will be adapted and extended so all options are covered. Including a `./pipeline/config.yml` file option and a JSON config file. 

Marking this conversation as resolved as different parameter naming is obsolete now.